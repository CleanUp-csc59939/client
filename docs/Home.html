<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Home</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Home</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Home<span class="signature">()</span><span class="type-signature"> &rarr; {Component}</span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="Home"><span class="type-signature"></span>new Home<span class="signature">()</span><span class="type-signature"> &rarr; {Component}</span></h4>
    

    



<div class="description">
    <p>return home page with multiple user events types*</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_homepage_home.jsx.html">components/homepage/home.jsx</a>, <a href="components_homepage_home.jsx.html#line22">line 22</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Return Home page showing the user's events</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Component</span>


    </dd>
</dl>

    





    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="~EventCarousel"><span class="type-signature">(inner) </span>EventCarousel<span class="signature">()</span><span class="type-signature"> &rarr; {HTMLBodyElement}</span></h4>
    

    



<div class="description">
    <p>Creates a carousel of events</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_homepage_home.jsx.html">components/homepage/home.jsx</a>, <a href="components_homepage_home.jsx.html#line39">line 39</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Returns the HTML code for user's event's in the carousel</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">HTMLBodyElement</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Banner.html">Banner</a></li><li><a href="CreateMeetUp.html">CreateMeetUp</a></li><li><a href="Header-CreateEvent.html">CreateEvent</a></li><li><a href="Home.html">Home</a></li><li><a href="Login.html">Login</a></li><li><a href="SingleEvent.html">SingleEvent</a></li></ul><h3>Global</h3><ul><li><a href="global.html#authHeader">authHeader</a></li><li><a href="global.html#ConvertDate">ConvertDate</a></li><li><a href="global.html#CustomSearch">CustomSearch</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.7</a> on Tue May 25 2021 00:40:39 GMT-0400 (Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>