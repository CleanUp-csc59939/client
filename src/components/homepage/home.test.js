import React from 'react';
import { shallow } from 'enzyme';
import Home from './home';

const testUser = {
  userID: 15,
  name: null,
  bio: null,
  location: null,
  img: 'https://res.cloudinary.com/cleanup/image/upload/v1621121500/download_remvsp.png',
};

describe('Create Event', () => {
  it('should render correctly in "debug" mode', async () => {
    const component = shallow(<Home currentUser={testUser} debug />);
    expect(component).toMatchSnapshot();
  });
});
