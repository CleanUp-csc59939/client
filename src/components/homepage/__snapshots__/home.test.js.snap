// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Event should render correctly in "debug" mode 1`] = `
<div
  id="loading"
>
  <img
    alt="Loading..."
    id="loading-image"
    src="loading/loading-gif.gif"
  />
</div>
`;
